lib/Auth/Process/COmanageDbClient.php

```diff
@@ -1282,6 +1136,135 @@ private function getTermsAgreement($personId, $coTermsId)
+                        }, ARRAY_FILTER_USE_KEY);
+                        if (!empty($invalid_keys) ) {
+                            SimpleSAML_Logger::error("[attrauthcomanage] Configuration error:'" . $req_attr . "' key configuration error");
+                            throw new SimpleSAML_Error_Exception(
+                                "attrauthcomanage configuration error:'" . $req_attr . "' key configuration error");
+                        }
+
+                    }
+                    if(array_key_exists($validation_rules['value_filter'])) {
+                        $invalid_values = array();
+                        foreach ($config[$req_attr] as $value) {
+                            if(!filter_var($value, $validation_rules['value_filter'])) {
+                                $invalid_values[] = $value;
+                            }
+                        }
+                        if (!empty($invalid_values) ) {
+                            SimpleSAML_Logger::error("[attrauthcomanage] Configuration error:'" . $req_attr . "' invalid value");
+                            throw new SimpleSAML_Error_Exception(
+                                "attrauthcomanage configuration error:'" . $req_attr . "' invalid value");
+                        }
+
+                    }
+
+                    $this->$req_attr=$config[$req_attr];
+                }
+            } elseif ($req_opt_key === 'optional') {
+                /*
+                 *  OPTIONAL ATTRIBUTES
+                 */
+                foreach($validation_list as $opt_attr => $type) {
+                    if (array_key_exists($opt_attr, $config)) {
+                        if (!$type($config[$opt_attr])) {
+                            SimpleSAML_Logger::error("[attrauthcomanage] Configuration error: " . $opt_attr . " not of type " . $type);
+                            throw new SimpleSAML_Error_Exception(
+                                "attrauthcomanage configuration error: $opt_attr not a boolean");
+                        }
+                        $this->$opt_attr = $config[$opt_attr];
+                    }
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Array of validation rules for the config params.
+     * Make changes here and not in the constructor
+     *
+     * @return string[][]
+     */
+    private function validateConfigParamRules() {
+        return array(
+            'required' => array(
+                'coId' => array(
+                    'type' => 'is_int'
+                ),
+                'urnNamespace' => array(
+                    'type' => 'is_string'
+                ),
+                'voRoles' => array(
+                    'type' => 'is_array'
+                ),
+                'urnAuthority' => array(
+                    'type' => 'is_string'
+                ),
+                'registryUrls' => array(
+                    'type' => 'is_array',
+                    'key_list' => array(
```

Comment:
`key_list` should be handled as the required list of keys

Reply:
done